### PR TITLE
Adding CS for Nesting Levels

### PR DESCRIPTION
**Pull request by votemike**
I have noticed some PRs with large functions and deeply nested logic. Rather than lots of comments asking for these to be refactored, I wonder what people think of adding this code sniffer. It should reduce the number of comments about refactoring methods and should also make for more readable code. Please reject if you don't agree.

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

**Comment by Tasiobg** [comment]
I like to keep a low nesting lever by separating bits of logic into different functions when possible so you can understand what is going on at a glance but I'm not 100% sure of the impact of enforcing anything so I can't support or oppose to enforcing.

I support the warning message but not sure about enforcing.

I'm interested in knowing what the other team members thing.

**Comment by felipeparaujo** [comment]
I agree with @Tasiobg, a warning is very useful, but enforcing might be a bit overreaching.

**Comment by felipeparaujo** [comment]
Also, before you merge this, make sure that showing a warning doesn't make the build process fail. Sometimes jenkins interprets warnings as errors.

File: src/Ds2013/Molecule/Image/ImagePresenter.php [hunk 1]
```diff
@@ -83,11 +83,7 @@ protected function validateOptions(array $options): void
         if (!is_array($options['srcsets'])) {
             throw new InvalidOptionException("Option 'srcsets' must be an array");
         } else {
-            foreach ($options['srcsets'] as $srcset) {
```
Comment (BPScott) [inline src/Ds2013/Molecule/Image/ImagePresenter.php]: Better refactoring here would be getting rid of the else clause as the throw acts as an early return

```
if (!is_array($options['srcsets'])) {
    throw new InvalidOptionException("Option 'srcsets' must be an array");
}

foreach ($options['srcsets'] as $srcset) {
    // ...
}

**Comment by votemike** [comment]
It seems that just a warning makes Jenkins fail.... Does that mean people don't want this sniff?

**Comment by felipeparaujo** [comment]
@votemike no, only that we'll have to update ci-docker. I believe that adding the -n argument when running phpcs would fix that. See here: https://manned.org/phpcs/a11ee962#head5

**Comment by Tasiobg** [comment]
In case someone is not aware, you can enable "Php code sniffer" and "Mess detector" within IntelliJ so it will warn you as you type. I found it a good practice

**Comment by BjornTwachtmann** [comment]
If you want to emit a warning on this, fine. It absolutely should not break the build. To be honest code sniffers with too many opinions about this kind of thing just piss me off. If I indent code deeply it's because I had a reason to...

**Comment by Tasiobg** [comment]
@BjornTwachtmann @votemike @felipeparaujo @BPScott 
This is an example of mess detector and code sniffer, those warning are quite soft and well integrated on IntelliJ.

screen shot 2017-05-22 at 11 31 42

**Pull request merged**